homeassistant/config_entries.py

```diff
@@ -1525,34 +1526,44 @@ def __setitem__(self, entry_id: str, entry: ConfigEntry) -> None:
         data[entry_id] = entry
         self._index_entry(entry)
 
+    def check_unique_id(self, entry: ConfigEntry) -> None:
+        """Check config entry unique id.
+
+        For hashable return and log warning
+        For non-hashable raise error
+        """
+        unique_id = entry.unique_id
+        if unique_id is not None and isinstance(unique_id, Hashable):
```

Comment:
The logic here is still wrong I think, causing component tests to fail.

Strings and integers are hashable so:
- None should pass
- str should pass
- other hashable should warn
- non hashable should raise


Non hashable should fail hard.
Hashable and not string should warn.
Hashable

Reply:
Re-reading the code on my PC (I was on mobile before), I think that string values are getting logged so maybe the test for integers should be parametrized to ensure that integers are logged and strings are NOT logged to prevent this from getting missed.

Reply:
I think that's a good proposal to expand the tests to cover the `pass` case (string)

Reply:
And also None case ;)